### PR TITLE
Add two files of GraphExplorer component in Master

### PR DESCRIPTION
**Pull request by Chris-MS-896**
Add two file
    "./src/Explorer/Graph/GraphExplorerComponent/EdgeInfoCache.ts",
    "./src/Explorer/Graph/GraphExplorerComponent/GraphData.ts",

### CODE REVIEW

File: src/Explorer/Graph/GraphExplorerComponent/GraphData.ts [hunk 1]
```diff
@@ -413,14 +413,14 @@ export class GraphData<V extends GremlinVertex, E extends GremlinEdge> {
...
-  public static getNodePropValue(node: D3Node, prop: string): string | number | boolean {
+  public static getNodePropValue(node: D3Node, prop: string): undefined | string | number | boolean {
     if (node.hasOwnProperty(prop)) {
       return (node as any)[prop];
     }
 
     // This is DocDB specific
-    if (node.hasOwnProperty("properties") && node.properties.hasOwnProperty(prop)) {
-      return node.properties[prop][0]["value"];
+    if (node.hasOwnProperty("properties") && node.properties!.hasOwnProperty(prop)) {
```
Comment (ato9000) [inline src/Explorer/Graph/GraphExplorerComponent/GraphData.ts:422]: send in a different PR and check with Steve and Laurent if we prefer the following style

    if (node && node.properties && node.properties.prop) {

**Review by ato9000 — CHANGES_REQUESTED** [review]

File: tsconfig.strict.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "noEmit": true,
     "strictNullChecks": true,
     "strict": true,
-    "noUnusedLocals": true
+    "noUnusedLocals": false
```
[the remark this answers is not in the document]
Comment (Chris-MS-896) [reply tsconfig.strict.json]: revert it

File: src/Explorer/Graph/GraphExplorerComponent/GraphData.ts (hunk 1 above)
Comment (Chris-MS-896) [reply src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: I chenged to node && node.properties && node.properties.prop &&, which has all properties in ndoe or not, but not 100% sure, will ask Steve and laurent

File: src/Explorer/Graph/GraphExplorerComponent/GraphData.ts [hunk 3]
```diff
@@ -413,13 +413,13 @@ export class GraphData<V extends GremlinVertex, E extends GremlinEdge> {
...
    */
-  public static getNodePropValue(node: D3Node, prop: string): string | number | boolean {
+  public static getNodePropValue(node: D3Node, prop: string): undefined | string | number | boolean {
     if (node.hasOwnProperty(prop)) {
       return (node as any)[prop];
     }
 
     // This is DocDB specific
-    if (node.hasOwnProperty("properties") && node.properties.hasOwnProperty(prop)) {
+    if (node && node.properties && node.properties.prop && node.properties.prop[0]["value"]) {
```
Comment (southpolesteve) [inline src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: You can also use optional chaining for this which is a bit shorter, bit I have no strong preference one way or the other https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Review by southpolesteve — APPROVED** [review]

File: src/Explorer/Graph/GraphExplorerComponent/GraphData.ts (hunk 1 above)
Comment (languy) [reply src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: I would opt for: `if (node?.properties?.prop) {`

File: src/Explorer/Graph/GraphExplorerComponent/GraphData.ts [hunk 4]
```diff
@@ -413,14 +413,14 @@ export class GraphData<V extends GremlinVertex, E extends GremlinEdge> {
...
+  public static getNodePropValue(node: D3Node, prop: string): undefined | string | number | boolean {
     if (node.hasOwnProperty(prop)) {
       return (node as any)[prop];
     }
 
     // This is DocDB specific
-    if (node.hasOwnProperty("properties") && node.properties.hasOwnProperty(prop)) {
-      return node.properties[prop][0]["value"];
+    if (node.hasOwnProperty("properties") && node.properties!.hasOwnProperty(prop)) {
+      return node.properties![prop][0]["value"];
```
Comment (languy) [inline src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: This should work. 
```suggestion
      return node.properties?prop[0]?value;
```

Comment (Chris-MS-896) [reply src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: image

Comment (Chris-MS-896) [reply src/Explorer/Graph/GraphExplorerComponent/GraphData.ts]: I chenged to "return node.properties?.prop[0]?.value;"

**Review by Chris-MS-896 — COMMENTED** [review]
I chenge it back, since those changes will have unit test false

**Review by ato9000 — APPROVED** [review]
